src/transport/SessionHandle.cpp

```diff
@@ -23,12 +23,59 @@ namespace chip {
 
 using namespace Transport;
 
+using AuthMode          = Access::AuthMode;
 using SubjectDescriptor = Access::SubjectDescriptor;
 
 SubjectDescriptor SessionHandle::GetSubjectDescriptor() const
 {
-    SubjectDescriptor subjectDescriptor = { .fabricIndex = mFabric };
-    // TODO: fill subject descriptor with proper fields
+    SubjectDescriptor subjectDescriptor;
+    if (IsSecure())
+    {
+        if (IsOperationalNodeId(mPeerNodeId))
+        {
+            // Currently ending up here after commissioning (good)
+            subjectDescriptor.authMode    = AuthMode::kCase;
+            subjectDescriptor.subject     = mPeerNodeId;
+            subjectDescriptor.fabricIndex = mFabric; // have fabric here
+            // TODO: add CATs
+            ChipLogDetail(DataManagement, "@@@ #####################################  CASE  %lx  (%u)", subjectDescriptor.subject,
+                          subjectDescriptor.fabricIndex);
+        }
+        else if (IsPAKEKeyId(mPeerNodeId))
+        {
+            // Should end up here during commissioning (but currently not)
+            subjectDescriptor.authMode    = AuthMode::kPase;
+            subjectDescriptor.subject     = mPeerNodeId;
+            subjectDescriptor.fabricIndex = mFabric; // also could have fabric here
+            ChipLogDetail(DataManagement, "@@@ #####################################  PASE  %lx  (%u)", subjectDescriptor.subject,
+                          subjectDescriptor.fabricIndex);
+        }
+        else if (mGroupId.HasValue())
+        {
+            // Should end up here during group messaging (how to test?)
+            subjectDescriptor.authMode = AuthMode::kGroup;
+            subjectDescriptor.subject  = kMinGroupNodeId | mGroupId.Value();
+            ChipLogDetail(DataManagement, "@@@ #####################################  Group  %lx", subjectDescriptor.subject);
+        }
+        else if (mPeerNodeId == kUndefinedNodeId)
+        {
+            // Currently ending up here during commissioning (bad, should be above)
+            subjectDescriptor.authMode    = AuthMode::kPase;
+            subjectDescriptor.subject     = kMinPAKEKeyId | mPeerNodeId;
+            subjectDescriptor.fabricIndex = mFabric;
+            ChipLogDetail(DataManagement, "@@@ #####################################  assumed-pase  (%u)",
+                          subjectDescriptor.fabricIndex);
+        }
+        else
+        {
+            // Should never end up here
+            ChipLogDetail(DataManagement, "@@@ #####################################  unknown  %lx", mPeerNodeId);
+        }
+    }
+    else
+    {
+        ChipLogDetail(DataManagement, "@@@ #####################################  not secure");
```

Comment:
Set `subjectDescriptor` to `AuthMode::kUnsecured` values or set them in default above.

Reply:
SubjectDescriptor does default to AuthMode::kNone.